ambry-replication/src/test/java/com/github/ambry/replication/ReplicationTest.java

```diff
@@ -436,6 +437,67 @@ public void onReplicaAddedOrRemovedCallbackTest() throws Exception {
     storageManager.shutdown();
   }
 
+  /**
+   * Test cluster map change callback in {@link ReplicationManager} for routing table updates.
+   * Test setup: When creating partitions, have one replica in LEADER state and rest in STANDBY states on each data center and
+   * later switch the states of replicas (LEADER to STANDBY and STANDBY to LEADER) on one of the DCs during the test
+   * Test condition: When replication manager receives onRoutingTableUpdate() indication after the remote replica states were updated,
+   * map of partition to peer leader replicas stored in replication manager should be updated correctly
+   * @throws Exception
+   */
+  @Test
+  public void onRoutingTableUpdateCallbackTest() throws Exception {
+    MockClusterMap clusterMap = new MockClusterMap();
+    ClusterMapConfig clusterMapConfig = new ClusterMapConfig(verifiableProperties);
+    MockHelixParticipant.metricRegistry = new MetricRegistry();
+    MockHelixParticipant mockHelixParticipant = new MockHelixParticipant(clusterMapConfig);
+    Pair<StorageManager, ReplicationManager> managers =
+        createStorageManagerAndReplicationManager(clusterMap, clusterMapConfig, mockHelixParticipant);
+    StorageManager storageManager = managers.getFirst();
+    MockReplicationManager replicationManager = (MockReplicationManager) managers.getSecond();
+    MockPartitionId existingPartition =
+        (MockPartitionId) replicationManager.partitionToPartitionInfo.keySet().iterator().next();
+    String currentDataCenter =
+        storageManager.getReplica(existingPartition.toString()).getDataNodeId().getDatacenterName();
+
+    //Trigger PartitionStateChangeListener callback to replication manager to notify that a local replica state has changed from STANDBY to LEADER
+    mockHelixParticipant.onPartitionBecomeLeaderFromStandby(existingPartition.toPathString());
```

Comment:
Just wonder how do we know `existingPartition` initially has a standby replica on local node?